### PR TITLE
EOL_HANDLING and OS detection with cmake

### PR DESCRIPTION
**Pull request by valco1994**
In this pull request, I present solutions for 2 problems:
 * It was declared somewhere that presence of separator ('|') at the end of the lines should be managed by macro `EOL_HANDLING`, but this functionality didn't work. It was solved by replacing `PR_STR` macro with `PR_VSTR_LAST` for the last attributes of the tables in print.c.
* There is macro `MACHINE` in the build script, which defines an underlying platform. Earlier, it was necessary to pass a value for this macro explicitly. Now I add automatic OS detection for Windows, Mac, and Linux in CMakeLists.txt. Probably, it should be developed further.

_Change in rnd.c can be ignored, it corresponds to the problem which is already solved using another approach._

### CODE REVIEW

File: rnd.c [hunk 1]
```diff
@@ -25,6 +25,8 @@ void
 dss_random(long *tgt, long lower, long upper, long stream)
 {
 	*tgt = UnifInt((long)lower, (long)upper, (long)stream);
+	if (stream < 0 || stream > MAX_STREAM)
+		stream = 0;
```
Comment (eyalroz) [inline rnd.c]: Why is this necessary? I mean, UnifInt() itself already does this. Is there some bug involved?

File: CMakeLists.txt [hunk 2]
```diff
@@ -3,16 +3,26 @@ cmake_minimum_required(VERSION 3.0)
 project(ssb-dbgen # SSB benchmark data (and query) generator utilities
 	LANGUAGES C)
 
+set(CMAKE_C_FLAGS_DEBUG "-g")
+set(CMAKE_C_FLAGS_RELEASE "-O3")
```
Comment (eyalroz) [inline CMakeLists.txt]: CMake adds this by default if you're building a Release build.

File: CMakeLists.txt [hunk 3]
```diff
@@ -3,16 +3,26 @@ cmake_minimum_required(VERSION 3.0)
 project(ssb-dbgen # SSB benchmark data (and query) generator utilities
 	LANGUAGES C)
 
+set(CMAKE_C_FLAGS_DEBUG "-g")
```
Comment (eyalroz) [inline CMakeLists.txt]: CMake adds this by default for the Debug build type. However, we _do_ need to make sure that a build type is indeed set.

File: CMakeLists.txt [hunk 4]
```diff
@@ -3,16 +3,26 @@ cmake_minimum_required(VERSION 3.0)
...
+elseif (APPLE)
+	set(MACHINE MAC CACHE STRING ${MACHINE_DOCSTRING})
+endif()
 set_property(CACHE MACHINE PROPERTY STRINGS ATT HP IBM ICL MVS SGI SUN U2200 VMS LINUX WIN32 MAC)
 
 set(WORKLOAD SSB CACHE STRING "Choice of benchmark / query workload (\"WORKLOAD\")")
 set_property(CACHE WORKLOAD PROPERTY STRINGS TPCH SSB)
 
-set(EOL_HANDLING OFF CACHE BOOL "Skip separator after the last field in generated lines?")
+set(EOL_HANDLING ON CACHE BOOL "Skip separator after the last field in generated lines?")
```
Comment (eyalroz) [inline CMakeLists.txt]: We should not change the default value for EOL_HANDLING - the original authors decided what the default is, and we stick with that...

File: CMakeLists.txt [hunk 5]
```diff
@@ -3,21 +3,21 @@ cmake_minimum_required(VERSION 3.0)
...
-if (UNIX AND NOT APPLE)
-	set(MACHINE "LINUX")
-endif()
-if (APPLE)
-	set(MACHINE "MAC")
+	set(MACHINE WIN32 CACHE STRING ${MACHINE_DOCSTRING})
+elseif (UNIX AND NOT APPLE)
+	set(MACHINE LINUX CACHE STRING ${MACHINE_DOCSTRING})
+elseif (APPLE)
+	set(MACHINE MAC CACHE STRING ${MACHINE_DOCSTRING})
```
Comment (eyalroz) [inline CMakeLists.txt]: I don't quite understand what's going on here. 

Now, I know it's actually not a good idea to have the user specify a machine string at all, and we should just detect the actual features, but given that this _is_ specified manually - why these changes?

... oh, I see, you're relying on CMake's internal variables. Interesting. However, I'm only willing to have that done when the user has not specifically set a value for `MACHINE`, to more closely follow original behavior. Also remember that not all Unix flavors are either Linux or Mac; for example - Sun, SGI and AT&T Unix.

**Comment by valco1994** [comment]
Ok, I'll make a separate PR for EOL_HANDLING

File: CMakeLists.txt (hunk 3 above)
Comment (eyalroz) [reply CMakeLists.txt]: ... and now I've done so.
